File: ruby/fizzbuzz.rb
def fizzbuzz(inputString)
    words = (splitOnWhitespace inputString)
    words = words.map.with_index { |word, i| fizzbuzzifyWord(word, i) }
    return words.join(" ")
  end
  
  def splitOnWhitespace(inputString) 
    inputString.gsub(/\s+/m, ' ').strip.split(" ")
  end
  
  def fizzbuzzifyWord (word, i)
      if (i+1) % 15 == 0
      return "fizzbuzz"
    end 
      if (i+1) % 5 == 0
      return "buzz"
    end
      if (i+1) % 3 == 0
      return "fizz"
    end 
  
  word
  end

File: typescript/fizzbuzz.ts

export default class Fizzbuzz {
  
  static fizzbuzz(inputString: string): string {
    
    return inputString
      .split(" ")
      .filter(item => item !== "")
      .map((word, i) => {
        if ((i + 1) % 15 === 0) return "fizzbuzz";
        if ((i + 1) % 5 === 0) return "buzz";
        if ((i + 1) % 3 === 0) return "fizz";
        return word;
      })
      .join(" ");
  };
  
}
